rasa/core/actions/action.py

```diff
@@ -1026,11 +1026,9 @@ async def _run_custom_action(
                 output_channel, nlg, tracker, domain
             )
             for event in custom_events:
-                if (
-                    isinstance(event, SlotSet)
-                    and tracker.get_slot(event.key) != event.value
-                ):
-                    slot_events.append(event)
+                if isinstance(event, SlotSet):
+                    if tracker.get_slot(event.key) != event.value:
+                        slot_events.append(event)
```

Comment:
Nite catch! Do we also need a test for this?

Reply:
It would probably be a good idea rather than relying on manual testing 😄 I'll create one now.